test/framework/http.go

```diff
@@ -199,11 +206,22 @@ func IsHTTPResponseArraySize(namespace, httpMethod, uri, path string, bodyFormat
 }
 
 // DoesHTTPResponseContain checks whether the response of an http request contains a certain string
-func DoesHTTPResponseContain(namespace, httpMethod, uri, path string, bodyFormat, bodyContent string, responseContent string) (bool, error) {
-	resultBody, err := ExecuteHTTPRequestWithStringResponse(namespace, httpMethod, uri, path, bodyFormat, bodyContent)
+func DoesHTTPResponseContain(namespace string, requestInfo HTTPRequestInfo, responseContent string) (bool, error) {
+	resultBody, err := ExecuteHTTPRequestWithStringResponse(namespace, requestInfo)
 	if err != nil {
 		return false, err
 	}
 
 	return strings.Contains(resultBody, responseContent), nil
 }
+
+// NewHTTPRequestInfo constructor creates a new HTTPRequestInfo struct
+func NewHTTPRequestInfo(httpMethod, uri, path, bodyFormat, bodyContent string) HTTPRequestInfo {
```

Comment:
What about providing separate function for GET and POST types?
For GET type you don't need to supply bodyFormat and bodyContent.

Reply:
Hmm, I was actually thinking about it, but just to get away with GET and POST strings. But your suggestion is even better, thanks!